src/main/java/io/stargate/sgv2/jsonapi/service/embedding/configuration/EmbeddingConfigSourceProvider.java

```diff
@@ -0,0 +1,23 @@
+package io.stargate.sgv2.jsonapi.service.embedding.configuration;
+
+import io.quarkus.runtime.annotations.StaticInitSafe;
+import io.smallrye.config.source.yaml.YamlConfigSource;
+import java.io.IOException;
+import java.net.URL;
+import java.util.Collections;
+import org.eclipse.microprofile.config.spi.ConfigSource;
+import org.eclipse.microprofile.config.spi.ConfigSourceProvider;
+
+@StaticInitSafe
+public class EmbeddingConfigSourceProvider implements ConfigSourceProvider {
+  @Override
```

Comment:
May be `EmbeddingProviderConfigSource` is better?

Reply:
Since it implements ConfigSourceProvider

Reply:
No, `ConfigSourceProvider` is from Quarkus, the meaning of that provider is not the same as our provider

Reply:
... I understand, that is why it is called SourceProvider.